api/src/pcapi/routes/backoffice/pro_users/blueprint.py

```diff
@@ -235,6 +239,33 @@ def delete(user_id: int) -> utils.BackofficeResponse:
     payload = DeleteBatchUserAttributesRequest(user_id=user.id)
     delete_user_attributes_task.delay(payload)
 
+    # Delete all related objects if the user has already been created as a beneficiary
+    beneficiary_import_status_list = (
+        beneficiary_import_status_models.BeneficiaryImportStatus.query.with_entities(
+            beneficiary_import_status_models.BeneficiaryImportStatus.id,
+        )
+        .join(
+            beneficiary_import_status_models.BeneficiaryImportStatus.beneficiaryImport,
+        )
+        .filter(
+            beneficiary_import_models.BeneficiaryImport.beneficiaryId == user_id,
+        )
+        .all()
+    )
+    if beneficiary_import_status_ids := [e[0] for e in beneficiary_import_status_list]:
+        beneficiary_import_status_models.BeneficiaryImportStatus.query.filter(
+            beneficiary_import_status_models.BeneficiaryImportStatus.id.in_(beneficiary_import_status_ids)
+        ).delete(
+            synchronize_session=False,
+        )
```

Comment:
```suggestion
    beneficiary_import_status_models.BeneficiaryImportStatus.query.filter(
        beneficiary_import_status_models.BeneficiaryImportStatus id.in_(
            beneficiary_import_status_models.BeneficiaryImportStatus.query.with_entities(
                beneficiary_import_status_models.BeneficiaryImportStatus.id,
            )
            .join(
                beneficiary_import_status_models.BeneficiaryImportStatus.beneficiaryImport,
            )
            .filter(
                beneficiary_import_models.BeneficiaryImport.beneficiaryId == user_id,
            )
        )
    ).delete(
        synchronize_session=False,
    )
```
on peut économiser un aller retour avec la db en passant par une subquery (a tester si on peut pas juste remplacer le .all par un .delete mais je crois pas que ce soit possible)